src/pages/book/[bookid]/[chapterid]/addSection.tsx

```diff
@@ -0,0 +1,36 @@
+"use client";
+import Head from "next/head";
+import { useRouter } from "next/router";
+import BookRenderer from "~/components/BookRenderer";
+
+import SectionForm from "~/components/SectionForm";
+export default function AddSectionPage() {
+  const router = useRouter();
+  const bookId = router.query.bookid as string; // Moved and type-checked
+  const chapterId = router.query.chapterid as string; // Moved and type-checked
+
+  if (!bookId || !chapterId) return <p>No book or chapter found!</p>; // Added check for both bookId and chapterId
```

Comment:
keep comments in the commit or the PR comments, code comments should only be semantic to the code, not about the meta-change